android/build.gradle

```
@@ -1,11 +1,74 @@
-// On top of your file import a JSON parser
 import groovy.json.JsonSlurper
 
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+
+String getPackageJsonPath() {
+    return findProperty("APPSFLYER_PACKAGE_JSON") ?: "$rootDir/../node_modules/appsflyer-capacitor-plugin/package.json"
+}
+
+def findNodeModulesDir(File currentDir) {
+    def dir = currentDir
+    while (dir != null) {
+        def nodeModulesDir = new File(dir, 'node_modules')
+        if (nodeModulesDir.exists()) {
+            return nodeModulesDir
+        }
+        dir = dir.parentFile
+    }
+    return null
+}
```

This code will find the first `node_modules` generally. However, in the monorepo structure of my company the first `node_modules` will be returned, so far so good, but it does not contain the `appsflyer-capacitor-plugin` package. My proposal for this method would be to extend its meaning to "find the first `node_modules` *containing the package* `packageName`".
```suggestion
def findNodeModulesDir(File currentDir, String packageName) {
    def dir = currentDir
    while (dir != null) {
        def nodeModulesDir = new File(dir, 'node_modules')
        if (nodeModulesDir.exists()) {
            def packageDir = new File(nodeModulesDir, packageName);
            if (packageDir.exists()) {
               return nodeModulesDir;
            }
        }
        dir = dir.parentFile
    }
    return null
}
```

---

Correct, it will find the first node modules directory from the base path, that's why we looking for an upper-level directory in line 56

---

Aah no I see. Thanks. 👍🏻